examples/ultrahdr_app.cpp

```diff
@@ -1302,6 +1317,67 @@ void UltraHdrAppInput::computeYUVSdrPSNR() {
   std::cout << "psnr yuv: \t" << mPsnr[0] << " \t " << mPsnr[1] << " \t " << mPsnr[2] << std::endl;
 }
 
+void UltraHdrAppInput::convertDestGamutRgba1010102() {
+  if (mDecodedUhdrRgbImage.cg == mHdrCg) return;
+
+  const float* coeffs = nullptr;
+  if (mDecodedUhdrRgbImage.cg == UHDR_CG_BT_709) {
+    if (mHdrCg == UHDR_CG_BT_2100)
+      coeffs = BT709ToBT2100RGB;
+    else if (mHdrCg == UHDR_CG_DISPLAY_P3)
+      coeffs = BT709ToP3RGB;
+  } else if (mDecodedUhdrRgbImage.cg == UHDR_CG_BT_2100) {
+    if (mHdrCg == UHDR_CG_BT_709)
+      coeffs = BT2100ToBT709RGB;
+    else if (mHdrCg == UHDR_CG_DISPLAY_P3)
+      coeffs = BT2100ToP3RGB;
+  } else if (mDecodedUhdrRgbImage.cg == UHDR_CG_DISPLAY_P3) {
+    if (mHdrCg == UHDR_CG_BT_709)
+      coeffs = P3ToBT709RGB;
+    else if (mHdrCg == UHDR_CG_BT_2100)
+      coeffs = P3toBT2100RGB;
+  }
+  if (coeffs == nullptr) {
+    std::cerr << "color matrix not present for gamut " << mDecodedUhdrRgbImage.cg
+              << " using BT2020Matrix" << std::endl;
+    return;
+  }
+
+  uint32_t* rgbData = static_cast<uint32_t*>(mDecodedUhdrRgbImage.planes[UHDR_PLANE_PACKED]);
+
+  for (size_t i = 0; i < mDecodedUhdrRgbImage.h; i++) {
+    for (size_t j = 0; j < mDecodedUhdrRgbImage.w; j++) {
+      float r0 = float(rgbData[mDecodedUhdrRgbImage.stride[UHDR_PLANE_PACKED] * i + j] & 0x3ff);
+      float g0 =
+          float((rgbData[mDecodedUhdrRgbImage.stride[UHDR_PLANE_PACKED] * i + j] >> 10) & 0x3ff);
+      float b0 =
+          float((rgbData[mDecodedUhdrRgbImage.stride[UHDR_PLANE_PACKED] * i + j] >> 20) & 0x3ff);
+
+      r0 /= 1023.0f;
+      g0 /= 1023.0f;
+      b0 /= 1023.0f;
+
+      float r = coeffs[0] * r0 + coeffs[1] * g0 + coeffs[2] * b0;
```

Comment:
Thank you for the change!

I suppose this should happen in linear space. I.e. mDecodedUhdrRgbImage is HLG for example, need to first convert it to linear, then convert gamut, then convert back to HLG

Reply:
Sorry, i missed this part. This change wont work at all.